### PR TITLE
mock search API

### PR DESCRIPTION
**Pull request by cameron1024**
adds expanded search capabilities to the mock

### CODE REVIEW

File: src/vit-testing/vitup/committee_1 [hunk 1]
```diff
@@ -1 +1 @@
-ed25519e_sk1hpx0pkpd3qu7ww9nx87y6vswhrk2lvey3ac3xzl4va6hw5rhapfz6sy48tywfnhq9u47arzc0fah7rlrlsvh22jhmxxdv8rv38n0evg3fzf55
+ed25519e_sk1pzjv5ghwf4mzvrqus0u7trplspuvq78pkzuwmz7s9cer6kq3vffv5xsyd4crgmvsm2mrnl0fmujgt8eavwjjtwwrtcdez23kq4gfvjcrnjhjg
```
Comment (dkijania) [inline src/vit-testing/vitup/committee_1:1]: I think we should discard all changes for 

src/vit-testing/vitup:
- block0.bin
- commitee_1
- genesis.yaml
- initial_setup.dot

File: src/vit-testing/vitup/voting_token.txt [hunk 2]
```diff
@@ -1 +1 @@
-[["Voter","00000000000000000000000000000000000000000000000000000000.b1eeb78a0f8907de6038847847efbede1d7a681b20566d068405381b4c2ef118"],["Representative","00000000000000000000000000000000000000000000000000000000.3c2eb3e0695c855d21f9441b5d0cd722bbc4ea0ede1412a320215fa629b99269"]]
\ No newline at end of file
+[["direct","00000000000000000000000000000000000000000000000000000000.fd700842aff361bcfa9fd18318034119754bd09a6344ad8dc59c4f1e86571926"],["rep","00000000000000000000000000000000000000000000000000000000.53c39b998a9cca56f75b031666b8a4ce48f50af1e2198126567f1782dbec5b46"]]
```
Comment (dkijania) [inline src/vit-testing/vitup/voting_token.txt:1]: And this too should be discarded

File: src/vit-testing/vitup/src/mode/mock/rest/search.rs [hunk 3]
```diff
@@ -73,74 +67,139 @@ async fn search_impl(
     };
     match table {
         Table::Challenges => {
```
Comment (minikin) [inline src/vit-testing/vitup/src/mode/mock/rest/search.rs:69]: @cameron 
Can we consider refactoring the body of the `search_impl` function to several functions, making it modular? 
It has more than 150 lines which make really hard to understand what's going on there.

**Review by minikin — APPROVED** [review]
Thank you, LGTM!

**Review by dkijania — APPROVED** [review]

**Pull request merged**